File: C#/Practice/LeetCode/Easy/Minimum Number Game.cs
// https://leetcode.com/problems/minimum-number-game/description/

public class Solution {
    public int[] NumberGame(int[] nums) {
    Array.Sort(nums);
    Array.Reverse(nums);
    List<int> resList = new List<int>();
    int first;
    int second;
    while (nums.Length > 0) {
        first = nums[nums.Length - 1];
        nums = nums[..^1];
        second = nums[nums.Length - 1];
        nums = nums[..^1];
        resList.Add(second);
        resList.Add(first);
    }
    return resList.ToArray();      
    }
}

File: TypeScript/Practice/LeetCode/Easy/Minimum Number Game.ts
// https://leetcode.com/problems/minimum-number-game/description/

function numberGame(nums: number[]): number[] {
    nums.sort((a, b) => b - a);
    const resArr: number[] = [];
    let first: number;
    let second: number;
    while (nums.length > 0) {
        first = nums.pop();
        second = nums.pop();
        resArr.push(second);
        resArr.push(first);
    }
    return resArr; 
};